location/src/main/java/org/motechproject/nms/location/domain/HealthFacility.java

```diff
@@ -0,0 +1,127 @@
+import java.util.List;
+
+@Entity(tableName = "nms_health_facilities")
+public class HealthFacility extends MdsEntity {
+
+    @Field
+    @UIDisplayable(position = 0)
+    @NotNull
+    private String name;
+
+    @Field
+    @Unique
+    @UIDisplayable(position = 1)
+    @NotNull
+    private Long code;
+
+    @Field
+    @UIDisplayable(position = 2)
+    @NotNull
+    private Integer healthFacilityType;
+
+    @Field
+    @UIDisplayable(position = 3)
+    @NotNull
+    private HealthBlock healthBlock;
+
+    @Field
+    @Cascade(delete = true)
+    @Persistent(mappedBy = "healthFacility", defaultFetchGroup = "true")
+    private List<HealthSubFacility> healthSubFacilities;
+
+    public HealthFacility() {
+        this.healthSubFacilities = new ArrayList<>();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Long getCode() {
+        return code;
+    }
+
+    public void setCode(Long code) {
+        this.code = code;
+    }
+
+    public Integer getHealthFacilityType() {
+        return healthFacilityType;
+    }
+
+    public void setHealthFacilityType(Integer healthFacilityType) {
+        this.healthFacilityType = healthFacilityType;
+    }
+
+    public HealthBlock getHealthBlock() {
+        return healthBlock;
+    }
+
+    public void setHealthBlock(HealthBlock healthBlock) {
+        this.healthBlock = healthBlock;
+    }
+
+    public List<HealthSubFacility> getHealthSubFacilities() {
+        return healthSubFacilities;
+    }
+
+    public void setHealthSubFacilities(List<HealthSubFacility> healthSubFacilities) {
+        this.healthSubFacilities = healthSubFacilities;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        HealthFacility that = (HealthFacility) o;
+
+        if (name != null ? !name.equals(that.name) : that.name != null) {
+            return false;
+        }
+        if (code != null ? !code.equals(that.code) : that.code != null) {
+            return false;
+        }
+        return !(healthFacilityType != null ? !healthFacilityType
+                .equals(that.healthFacilityType) : that.healthFacilityType != null);
+
+    }
+
+    @Override
+    public int hashCode() {
+        int result = name != null ? name.hashCode() : 0;
+        result = 31 * result + (code != null ? code.hashCode() : 0);
+        result = 31 * result + (healthFacilityType != null ? healthFacilityType.hashCode() : 0);
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return "HealthFacility{" +
+                "name='" + name + '\'' +
+                ", code=" + code +
+                ", healthFacilityType=" + healthFacilityType +
+                ", healthBlock=" + healthBlock +
+                '}';
```

Comment:
It was causing trouble.  I was getting stack overflow and detached member
exceptions.  Also the code is unique in the database so I was thinking that
is god for equality testing.
On Apr 27, 2015 6:17 PM, "Frank Huster" <notifications@github.com> wrote:

> In
> location/src/main/java/org/motechproject/nms/location/domain/HealthFacility.java
> <https://github.com/motech-implementations/mim/pull/192#discussion_r29207682>
> :
>
> > +    @Override
> > +    public int hashCode() {
> > +        int result = name != null ? name.hashCode() : 0;
> > +        result = 31 * result + (code != null ? code.hashCode() : 0);
> > +        result = 31 * result + (healthFacilityType != null ? healthFacilityType.hashCode() : 0);
> > +        return result;
> > +    }
> > +
> > +    @Override
> > +    public String toString() {
> > +        return "HealthFacility{" +
> > +                "name='" + name + '\'' +
> > +                ", code=" + code +
> > +                ", healthFacilityType=" + healthFacilityType +
> > +                ", healthBlock=" + healthBlock +
> > +                '}';
>
> Why should we be excluding collection data members from equals?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/pull/192/files#r29207682>.
>